### PR TITLE
fix(Launcher): dumpio bug

### LINKED ISSUE #2046: Using the dumpio setting throws an error in v1.1.0
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 1.1.0
* Platform / OS version: Ubuntu 16.04
* Node.js version: 6.13.0

**What steps will reproduce the problem?**
Running this code:
```
const puppeteer = require('puppeteer');

puppeteer.launch({
  dumpio: true
}).then(function (browser) {
  // irrelevant code here
});
```

**What is the expected result?**
This should not cause an error.

**What happens instead?**
There is a thrown error:
```
(node: 23455) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: input.on is not a function
```
And the program exits.

**More notes**
Version 1.0.0 seems not to have this issue.

### PR DESCRIPTION
**Pull request by yanivefraim**
fixes #2046

### CODE REVIEW

**Comment by yanivefraim** [comment]
I would like to use polling with a tiny util called [eventually](https://github.com/wix/wix-eventually), in order to replace this [part](https://github.com/GoogleChrome/puppeteer/pull/2071/files#diff-c1129c8b045390789fa8ff62f2c6b4a9R280) (it will be much cleaner). Is it ok to install it as a devDependency?

File: test/fixtures/dumpio.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+(async() => {
+  const [, , puppeteerRoot, options, emptyPage] = process.argv;
+  const browser = await require(puppeteerRoot).launch(JSON.parse(options));
+  const page = await browser.newPage();
+  await page.goto(emptyPage);
+  await page.evaluate(() => console.log('test'));
```
Comment (aslushnikov) [inline test/fixtures/dumpio.js]: nit: let's log `"MAGIC_DUMPIO_TEST"` to be distinct

File: test/test.js [hunk 2]
```diff
@@ -271,6 +271,19 @@ describe('Puppeteer', function() {
...
+      const path = require('path');
+      const options = Object.assign({dumpio: true}, defaultBrowserOptions);
+      const res = spawn('node',
+          [path.resolve('test', 'fixtures', 'dumpio.js'), PROJECT_ROOT, JSON.stringify(options), server.EMPTY_PAGE]);
+      const response = await new Promise(resolve => res.stderr.on('data', data => {
+        if (data.toString('utf8').includes('test'))
+          resolve(data.toString('utf8'));
+      }));
+
+      expect(response).toContain('test');
```
Comment (aslushnikov) [inline test/test.js]: you should await for the node script to terminate

**Review by aslushnikov — APPROVED** [review]
Impressive test, thank you

**Comment by aslushnikov** [comment]
> I would like to use polling with a tiny util called eventually, in order to replace this part (it will be much cleaner). Is it ok to install it as a devDependency?

Can we instead wait for the script to finish and check it's stderr after that?

File: test/fixtures/dumpio.js (hunk 1 above)
Comment (yanivefraim) [reply test/fixtures/dumpio.js]: Sure, done...

File: test/test.js (hunk 2 above)
Comment (yanivefraim) [reply test/test.js]: Done

**Comment by yanivefraim** [comment]
> > I would like to use polling with a tiny util called eventually, in order to replace this part (it will be much cleaner). Is it ok to install it as a devDependency?
> 
> Can we instead wait for the script to finish and check it's stderr after that?

Yes, sure. It was too late at night for me (: 
Done (thanks)

**Pull request merged**